Source/WarpX.cpp

```diff
@@ -340,6 +340,24 @@ WarpX::ReadParameters ()
     {
         ParmParse pp("warpx");
 
+        // set random seed
+        std::string random_seed = "default";
+        pp.query("random_seed", random_seed);
+        std::random_device rd;
+        std::uniform_int_distribution<int> dist(0, INT_MAX);
+        if ( random_seed != "default" ) {
+            unsigned long myproc_1 = ParallelDescriptor::MyProc() + 1;
+            if ( random_seed == "random" ) {
+                unsigned long seed = myproc_1 * dist(rd);
+                ResetRandomSeed(seed);
+            } else if ( std::stoi(random_seed) > 0 ) {
+                unsigned long seed = myproc_1 * std::stoul(random_seed);
+                ResetRandomSeed(seed);
+            } else {
+                Abort("warpx.random_seed must be \"random\" or a number.");
```

Comment:
For completeness, this could be changed to
`Abort("warpx.random_seed must be \"default\", \"random\" or an integer > 0.");`